connectors/sources/confluence.py

```diff
@@ -312,8 +341,230 @@ def get_default_configuration(cls):
+            return False
+
+        return self.configuration["use_document_level_security"]
+
+    def access_control_query(self, access_control):
+        # filter out 'None' values
+        filtered_access_control = list(
+            filter(
+                lambda access_control_entity: access_control_entity is not None,
+                access_control,
+            )
+        )
+
+        return {
+            "query": {
+                "template": {"params": {"access_control": filtered_access_control}},
+                "source": {
+                    "bool": {
+                        "filter": {
+                            "bool": {
+                                "should": [
+                                    {
+                                        "bool": {
+                                            "must_not": {
+                                                "exists": {"field": ACCESS_CONTROL}
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "terms": {
+                                            f"{ACCESS_CONTROL}.enum": filtered_access_control
+                                        }
+                                    },
+                                ]
+                            }
+                        }
+                    }
+                },
+            }
+        }
+
+    def _decorate_with_access_control(self, document, access_control):
+        if self._dls_enabled():
+            document[ACCESS_CONTROL] = list(
+                set(document.get(ACCESS_CONTROL, []) + access_control)
+            )
+
+        return document
+
+    async def _user_access_control_doc(self, user):
+        """Generate a user access control document.
+
+        This method generates a user access control document based on the provided user information.
+        The document includes the user's account ID, prefixed account ID, prefixed account name,
+        a set of prefixed group IDs, and a set of prefixed role keys. The access control list is
+        then constructed using these values.
+
+        Args:
+            user (dict): A dictionary containing user information, such as account ID, display name, groups, and application roles.
+
+        Returns:
+            dict: A user access control document with the following structure:
+                {
+                    "_id": <account_id>,
+                    "identity": {
+                        "account_id": <_prefixed_account_id>,
+                        "display_name": <_preffixed_account_name>
+                    },
+                    "created_at": <iso_utc_timestamp>,
+                    ACCESS_CONTROL: [<_prefixed_account_id>, <_prefixed_group_ids>, <_prefixed_role_keys>]
+                }
+        """
+        account_id = user.get("accountId")
```

Comment:
This whole method looks very similar to the one in the Jira connector. Should we create a common `atlassian.py` module to be able to reuse this functionality?

Reply:
Yes this was in my mind to move common code in `atlassian.py`(Already created this common module during Advanced Sync Rules implementation) once both DLS implementation will merge we will raise one incremental PR for the same 👍

Reply:
Thanks! Can you create a corresponding issue so we can track it?

Reply:
Yes @moxarth-elastic created it https://github.com/elastic/connectors-python/issues/1417